internal/provider/jwt_issuers_test.go

```diff
@@ -0,0 +1,364 @@
+/*
+ Copyright 2024 The Cockroach Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/v2/pkg/client"
+	mock_client "github.com/cockroachdb/terraform-provider-cockroach/mock"
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	testIssuerURL = "https://issuer-%s.com"
+
+	testAudience        = "audience"
+	testAudienceUpdated = "audience_updated"
+
+	testJWKS        = "{}"
+	testJWKSUpdated = "{updated: 1}"
+
+	testClaim        = "sub"
+	testClaimUpdated = "email"
+
+	testTokenID1 = "token_id1"
+	testCCID1    = "cc_id1"
+	testIsRegex1 = false
+
+	testTokenID2 = "token_id2"
+	testCCID2    = "cc_id2"
+	testIsRegex2 = true
+
+	testIdentityMap = []IdentityMapEntry{
+		{
+			TokenIdentity: types.StringValue(testTokenID1),
+			CcIdentity:    types.StringValue(testCCID1),
+			IsRegex:       types.BoolValue(testIsRegex1),
+		},
+		{
+			TokenIdentity: types.StringValue(testTokenID2),
+			CcIdentity:    types.StringValue(testCCID2),
+			IsRegex:       types.BoolValue(testIsRegex2),
+		},
+	}
+	testIdentityMapUpdated = []IdentityMapEntry{
+		{
+			TokenIdentity: types.StringValue(testTokenID1),
+			CcIdentity:    types.StringValue(testCCID1),
+			IsRegex:       types.BoolValue(testIsRegex1),
+		},
+	}
+)
+
+// TestAccJWTIssuerResource attempts to create, check, and destroy
+// a real JWT Issuer. It will be skipped if TF_ACC isn't set.
+// In order to work, the `JwtIssuerEnabled` Feature Flag must be enabled and
+// the test org must have Org SSO enabled (no need for SAML/OIDC).
```

Comment:
It would be nice if we could enable this via terraform.  @marksoper do you think this is something customers would want?

Reply:
@fantapop what specifically is the feature you're referring to?

Reply:
In this case, I'm referring to being able to enable cloud SSO.   But more generally it seems like there might be a set of org configurations that we could consider allowing for edit.  I see that Org name is allowed to be updated as well.

<img width="1739" alt="Screenshot 2024-09-23 at 10 56 55 AM" src="https://github.com/user-attachments/assets/1551ce47-6904-49fd-8d4c-ce6a4376bdc3">
<img width="1690" alt="Screenshot 2024-09-23 at 10 57 52 AM" src="https://github.com/user-attachments/assets/536c163d-f843-4733-8be0-e90a82454f2b">

Reply:
@fantapop My initial thought is to wait until we have customer demand for doing this. Org level features aren't an obvious fit for Terraform:
- e.g. the org name being under Terraform when there's no org resource
- Enabling SSO is likely a one-time change. If there's configuration associated with SSO that needs to be managed then it would be a good fit. A quick check of customer signal doesn't turn up anything.

Reply:
If we did this, we'd add an Org level resource.   Ack on customer signal .  I was thinking it might be nice to be able to set this up inside the test but I realize it would probably be a painful because the only org we have to access to is the org that's running the tests.